microprot/scripts/split_search.py

```diff
@@ -453,6 +455,53 @@ def frag_size(hit):
     return len(subseq) - subseq.count('-')
 
 
+def correct_header_positions(header, new_start, new_end, out=sys.stderr):
+    """Prevents chaining of start-end positions.
+
+    Parameters
+    ----------
+    header : str
+        The existing fasta header string with start-end positions as a suffix.
+        Valid header format is specified as the regular expression:
+        '(.+)_(\d+)\-(\d+)$'
+        Examples for valid header: 'test_1_251-330', 'NZ_GG666849.1_2_251-330'
+        Examples for invalid headers: 'NZ_GG666849.1_2', 'test_1_251:330'
+    new_start : int
+        The new start position.
+    new_end : int
+        The new end position.
+    out : file handle
+        File handle into verbosity information should be printed.
+
+    Returns
+    -------
+    str : the new fasta header with updated start, end positions.
+
+    Raises
+    ------
+    ValueError if
+        a) input header string does not follow expected format
+        b) old and new start,end positions are incompatible."""
```

Comment:
PEP8 recommends that a docstring ends with """ in a new line. Also there should be a whitespace after `,`.